src/parsers/route.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright 2020-present Daniel Trugman
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "pfs/parsers.hpp"
+#include "pfs/utils.hpp"
+#include "pfs/types.hpp"
+
+#include <set>
+#include <sstream>
+
+namespace pfs {
+namespace impl {
+namespace parsers {
+
+route parse_route_line(const std::string& line)
+{
+    // Some examples:
+    // clang-format off
+    // Iface   Destination     Gateway         Flags   RefCnt  Use     Metric  Mask            MTU     Window  IRTT
+    // eth0    00000000        07030301        0003    0       0       0       00000000        0       0       0
+    // eth0    07030301        00000000        0001    0       0       0       00F0FFFF        0       0       0
+    route route;
+
+    static const char DELIM = '\t';
+    auto tokens = utils::split(line, DELIM);
+
+    if (tokens.size() != 11)
+    {
+        throw parser_error("Corrupted route - Unexpected tokens count", line);
+    }
+
+    // Parse tokens into route struct
+    route.iface          = tokens[0];
+    route.destination    = pfs::ip(pfs::ipv4(std::stoul(tokens[1], nullptr, 16)));
+    route.gateway        = pfs::ip(pfs::ipv4(std::stoul(tokens[2], nullptr, 16)));
+    route.flags          = std::stoul(tokens[3], nullptr, 16);
```

Comment:
Please use `utils::stot`, it provides a cleaner API for decimals and hexadecimals and additional checks.